modules/vmfcore/test/test_compressor.cpp

```diff
@@ -0,0 +1,235 @@
+/*
+ * Copyright 2015 Intel(r) Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http ://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include "test_precomp.hpp"
+
+using namespace vmf;
+
+//Bad compressor that increases the size of input data
+class BloatingCompressor : public vmf::Compressor
+{
+public:
+    static const int factor = 5;
+
+    virtual void compress(const vmf::vmf_string& input, vmf::vmf_rawbuffer& output)
+    {
+        output.clear();
+        for(auto it = input.begin(); it != input.end(); ++it)
+        {
+            for(int i = 0; i < factor; i++)
+                output.push_back(*it);
+        }
+    }
+
+    virtual void decompress(const vmf::vmf_rawbuffer& input, vmf::vmf_string &output)
+    {
+        output.clear();
+        for(auto it = input.begin(); it != input.end();)
+        {
+            char c = *it++;
+            for(int i = 1; i < factor; i++)
+            {
+                char cc = *it++;
+                if(cc != c)
+                {
+                    throw std::runtime_error("Incorrect input to bloating compressor");
+                }
+            }
+            output.push_back(c);
+        }
+    }
+
+    std::shared_ptr<Compressor> createNewInstance() const
+    {
+        return std::shared_ptr<Compressor>(new BloatingCompressor);
+    }
+
+    virtual vmf::vmf_string getId()
+    {
+        return "com.intel.vmf.compressor.test.bloating";
+    }
+};
+
+
+class TestCompressor : public ::testing::TestWithParam<std::string>
+{
+protected:
+    void SetUp()
+    {
+        vmf::initialize();
+        //register bloating compressor
+        std::shared_ptr<Compressor> bloating = std::make_shared<BloatingCompressor>();
+        vmf::Compressor::registerNew(bloating);
+    }
+
+    void TearDown()
+    {
+        vmf::Compressor::unregister("com.intel.vmf.compressor.test.bloating");
+        vmf::terminate();
+    }
+
+    std::string generateData(int nChars)
+    {
+        std::string result;
+        srand(0); result.reserve(nChars);
+        for(int i = 0; i < nChars; i++)
+        {
+            result.push_back( (char)(rand()%256) );
+        }
+        return result;
```

Comment:
this can be done better via:
```cpp
        std::string result(nChars, 0);
        std::generate_n(result.begin(), nChars, []()->char {return (rand() % 254) + 1;});
```